### DISCUSSION TITLE
HelmRepository Chart Version set from configMap

### DISCUSSION THREAD

**Discussion by StickNitro** [fluxcd/flux2#4441] [Q&A]
Is it possible to set the `HelmRelease.spec.chart.spec.chart.version` from a ConfigMap?  

I am working on a project and we would like to be able to set this value from a ConfigMap. I am aware that we can use semantic version ranges but in our use case we want to pin the version of the chart through a ConfigMap that would be updated by the service pipeline dependent on the maturity level of the service.

For example:

    apiVersion: helm.toolkit.fluxcd.io/v2beta1
    kind: HelmRelease
    metadata:
      name: platform-foo
      namespace: flux-system
    spec:
      interval: 5m
      chart:
        spec:
          chart: helm/foo-platform-bar
          version: "{{ .Values.buildVariables.chartVersion }}"
          sourceRef:
            kind: HelmRepository
            name: environment-helm-repository
            namespace: flux-system
          interval: 1m
      valuesFrom:
        - kind: ConfigMap
          name: environment-build-variables
          valuesKey: values.json
          targetPath: buildVariables

**Comment by stefanprodan** [comment]
You can do `version: "${app_version}"` and use a ConfigMap in the Flux Kustomization that manages this HelmRelease. 

Docs here: https://fluxcd.io/flux/components/kustomize/kustomizations/#post-build-variable-substitution

**Reply by MartinEmrich** [reply]
Same question: Install the same stack on multiple clusters, with the only degree-of-freedom being the version number specs.

Here, there's no Kustomization to manage the HelmRelease, and also no Repository accessible to Flux to host manifests for one. In fact Flux2 was quite revolutionary for us, helping us getting rid of Kustomizations or other direct Git repository integrations. It's HelmReleases only from now on.

I could of course create a "meta" Helm Chart containing exactly that HelmRelease and template the inner chart version from a Helm Value taken from the configmap, but that also seem a bit over-the-top.

An altenative idea: instead of taking the version from the Values (Which could also be tricky, what if the chart set's its own version Value itself?), take it directly from a ConfigMap:

````
spec:
  chart:
    spec:
      chart: helm/foo-platform-bar
      versionRef:
        configMapKeyRef:
          name: my-version-configmap
          key: foo-platform-bar-version
      sourceRef:
        kind: HelmRepository
````

Is there something like that already possible?

**Reply by MartinEmrich** [reply]
Oh no, googled again, found my own question 1,5y later :/
